includes/discovery/loadbalancers/f5-ltm.inc.php

```diff
@@ -0,0 +1,272 @@
+$tblBigIP = array();
+
+// Let's gather some data..
+$ltmVirtualServEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.10.1.2.1', 0);
+$ltmVsStatusEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.10.13.2.1', 0);
+$ltmPoolEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.5.1.2.1', 0);
+$ltmPoolMemberEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.5.3.2.1', 0);
+$ltmPoolMbrStatusEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.5.6.2.1', 0);
+
+/*
+ * False == no object found - this is not an error, OID doesn't exist.
+ * null  == timeout or something else that caused an error, OID may exist but we couldn't get it.
+ */
+if (!is_null($ltmVirtualServEntry) || !is_null($ltmVsStatusEntry) || !is_null($ltmPoolEntry) || !is_null($ltmPoolMemberEntry) || !is_null($ltmPoolMbrStatusEntry)) {
+    // No Nulls, lets go....
+    d_echo("Objects Found:\n");
+
+    // Process the Virtual Servers
+    foreach ($ltmVsStatusEntry as $oid => $value) {
+        $result = array();
+
+        // Find all Virtual server names and UID's, then we can find everything else we need.
+        if (strpos($oid, '1.3.6.1.4.1.3375.2.2.10.13.2.1.1.') !== false) {
+            list($null, $index) = explode('1.3.6.1.4.1.3375.2.2.10.13.2.1.1.', $oid);
+            $result['type'] = 'f5-ltm-vs';
+            $result['UID'] = (string)$index;
+            $result['label'] = $value;
+            // The UID is far too long to have in a RRD filename, use a hash of it instead.
+            $result['hash'] = hash('crc32', $result['UID']);
+
+            // Now that we have our UID we can pull all the other data we need.
+            $result['IP'] = hex_to_ip($ltmVirtualServEntry['1.3.6.1.4.1.3375.2.2.10.1.2.1.3.'.$index]);
+            $result['port'] = $ltmVirtualServEntry['1.3.6.1.4.1.3375.2.2.10.1.2.1.6.'.$index];
+            $result['pool'] = $ltmVirtualServEntry['1.3.6.1.4.1.3375.2.2.10.1.2.1.19.'.$index];
+
+            // 0 = None, 1 = Green, 2 = Yellow, 3 = Red, 4 = Blue
+            $result['state'] = $ltmVsStatusEntry['1.3.6.1.4.1.3375.2.2.10.13.2.1.2.'.$index];
+            if ($result['state'] == 2) {
+                // Looks like one of the VS Pool members is down.
+                $result['status'] = 1;
+                $result['error'] = $ltmVsStatusEntry['1.3.6.1.4.1.3375.2.2.10.13.2.1.5.'.$index];
+            } elseif ($result['state'] == 3) {
+                // Looks like ALL of the VS Pool members is down.
+                $result['status'] = 2;
+                $result['error'] = $ltmVsStatusEntry['1.3.6.1.4.1.3375.2.2.10.13.2.1.5.'.$index];
+            } else {
+                // All is good.
+                $result['status'] = 0;
+                $result['error'] = '';
+            }
+        }
+
+        // Do we have any results
+        if (count($result) > 0) {
+            // Let's log some debugging
+            d_echo("\n\n".$result['type'].": ".$result['label']."\n");
```

Comment:
All of these d_echo()'s (in all the code below as well) is a bit much, can you condense it down or remove unnecessary debug info